moveit_ros/benchmarks/src/RunBenchmark.cpp

```diff
@@ -56,17 +56,65 @@ int main(int argc, char** argv)
   rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("moveit_run_benchmark", node_options);
 
   // Read benchmark options from param server
-  moveit_ros_benchmarks::BenchmarkOptions opts(node);
+  moveit_ros_benchmarks::BenchmarkOptions options(node);
   // Setup benchmark server
   moveit_ros_benchmarks::BenchmarkExecutor server(node);
 
   std::vector<std::string> planning_pipelines;
-  opts.getPlanningPipelineNames(planning_pipelines);
-  server.initialize(planning_pipelines);
+  options.getPlanningPipelineNames(planning_pipelines);
+  if (!server.initialize(planning_pipelines))
+  {
+    RCLCPP_ERROR(LOGGER, "Failed to initialize benchmark server.");
+    rclcpp::shutdown();
+    return 1;
+  }
 
-  // Running benchmarks
-  if (!server.runBenchmarks(opts))
-    RCLCPP_ERROR(LOGGER, "Failed to run all benchmarks");
+  if (options.scene_name_.empty())
+  {
+    std::vector<std::string> scene_names;
+    try
+    {
+      warehouse_ros::DatabaseLoader db_loader(node);
+      warehouse_ros::DatabaseConnection::Ptr warehouse_connection = db_loader.loadDatabase();
```

Comment:
There doesn't seem to be any resource handling for `warehouse_connection` or `planning_scene_storage`. Should this be a unique_ptr instead? Can `planning_scene_storage` be an object instead of a pointer?

Reply:
No, because `boost::shared_ptr<DatabaseConnection> Ptr` we need to  clean this up first :/